Mathlib/Data/Fin/Tuple/Sort.lean

```diff
@@ -107,6 +108,20 @@ theorem monotone_sort (f : Fin n → α) : Monotone (f ∘ sort f) := by
   exact (monotone_proj f).comp (graphEquiv₂ f).monotone
 #align tuple.monotone_sort Tuple.monotone_sort
 
+/-- All the elements `· ≤ a` appear the start of a sorted tuple -/
+theorem sort_lt_at_start_of_monotone {α} [LinearOrder α] (m : ℕ) (f : Fin m → α) (a : α)
+    (h_sorted : Monotone f)
```

Comment:
We should have the `Antitone` variant too which just swaps the arguments of the `le` in the appropriate places. If you're lucky it follows immediately from `OrderDual`.